db/db_impl/db_impl.cc

```diff
@@ -1496,6 +1496,28 @@ bool DBImpl::SetPreserveDeletesSequenceNumber(SequenceNumber seqnum) {
   }
 }
 
+Status DBImpl::GetFullHistoryTsLow(ColumnFamilyHandle* column_family,
+                                   std::string* ts_low) {
+  if (ts_low == nullptr) {
+    return Status::InvalidArgument("Input ts_low is nullptr");
+  }
+  ColumnFamilyData* cfd;
+  if (column_family == nullptr) {
+    cfd = default_cf_handle_->cfd();
+  } else {
+    auto cfh = static_cast_with_check<ColumnFamilyHandleImpl>(column_family);
+    assert(cfh != nullptr);
+    cfd = cfh->cfd();
+  }
+  if (cfd->user_comparator()->timestamp_size() == 0) {
+    return Status::NotSupported(
```

Comment:
I realize I mentioned "we can return either NotSupported() or InvalidArgument()". I gave it another thought, and now I am leaning more towards returning `InvalidArgument()` here. Sorry for the ambiguity. We do support this operation, it's just we are passing a column family as argument that does not enable timestamp, which seems a case of invalid argument.

Reply:
fixed